### DISCUSSION TITLE
Textlog implementation for UVM profiling

### DISCUSSION THREAD

**Discussion by DavidPoliakoff** [LLNL/Caliper#350] [Q&A]
So, we've done UVM profiling at end of run, but now we have a user who has a use case in which the program doesn't terminate. They want to see where the UVM is thrashing and evicting their pages. I thought of a few implementations for this, but the most extensible is to use the textlog service.

Our report-based config:

```
KOKKOS_PROFILE_LIBRARY=/path/to/libcaliper.so CALI_CALIPER_ATTRIBUTE_DEFAULT_SCOPE=process CALI_LOG_VERBOSITY=1 CALI_SERVICES_ENABLE=cuptitrace:cupti:recorder:timestamp:event:kokkoslookup:alloc:aggregate:kokkostime:report CALI_REPORT_CONFIG="SELECT function,alloc.label#cupti.fault.addr,cupti.uvm.kind,inclusive_sum(sum#cupti.uvm.bytes),sum(sum#cupti.uvm.bytes) WHERE sum#cupti.uvm.bytes GROUP BY function,alloc.label#cupti.fault.addr,cupti.uvm.kind FORMAT TABLE" CALI_ALLOC_RESOLVE_ADDRESSES=TRUE CALI_CUPTITRACE_ACTIVITIES=uvm CALI_CUPTITRACE_FLUSH_ON_SNAPSHOT=true $@
```

I've tried some obvious alterations (add textlog service, make it trigger on cupti.sync, make the textlog_formatstring print "%function% %alloc.label#cupti.fault.addr%") but none of them work. Any thoughts? Basically I want each cupti.sync to print the associated UVM movements

**Comment by daboehme** [comment]
The textlog service currently only triggers on events created by the event service. This should be easy to fix, but I don't have a quick workaround unfortunately.

**Comment by daboehme** [comment]
So with the latest commit (326dae0) this should work if you set the textlog trigger to `cupti.uvm.kind`.
